src/op_handlers/mul.rs

```diff
@@ -0,0 +1,8 @@
+use crate::address_operands::{address_operands_read, address_operands_store};
+use crate::{opcode::Opcode, state::VMState};
+
+pub fn _mul(vm: &mut VMState, opcode: Opcode) {
+    let (src0, src1) = address_operands_read(vm, &opcode);
+    let res = src0 * src1;
```

Comment:
These are 256 bit numbers, keep in mind the spec specifies the mul instruction multiplies the two numbers modulo 2**512. 
The high and low bits are also returned in 2 separate operands, I think address_operands_read only takes into account the first operand destination, correct me If I'm wrong.
Also, we should check if overflow has occurred with mul_overflow, check #25 for an example on how to do this.

Ideally, this PR should also implement the logic for the flag changes, if specified. You can also check #25 for an example.

Reply:
You are right, I'll change the function, it's corresponding test and check overflow.

Should we also implement the logic for flag changes in this PR, or leave it for a separate one as it's been done with #25?

Reply:
Yes, bring the changes from #25 now that is merged.